### ISSUE TITLE
Interface Display Template of TreeView not showing configured relative values within Drag'n'Drop Item Preview.

### ISSUE THREAD

**Issue by calbrecht** [directus/directus#9511]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes my problem.

### Describe the Bug

Within Editing Item View of Collection in the drag'n'drop children item preview [1], the relative values (configured in Interface with Tree View [2]) are not shown. Instead the id of the collection item is shown [1].

[1] ![image](https://user-images.githubusercontent.com/1516457/140482714-68639a08-90c2-4f16-96ec-87cc70f7f495.png)
[2] ![image](https://user-images.githubusercontent.com/1516457/140482596-4dfd2cf6-dbbc-4543-9458-c0760ea575c2.png)

### To Reproduce

* Create Collection `TreeView`
* Add String Field `Label`
* Add TreeView Field `Children`
* Change Interface of Field `Children` to `TreeView`
* Choose `Label` in Interface Display Template
* Create Collection Item in Collection `TreeView`
* Create Child Collection Item in Collection `TreeView`
* Observe the Interface Display Template showing `id` value instead of `Label`

### Errors Shown

NA

### What version of Directus are you using?

main HEAD 9.0.0

### What version of Node.js are you using?

v16.13.0

### What database are you using?

Postgres 12

### What browser are you using?

Firefox

### What operating system are you using?

NixOS

### How are you deploying Directus?

Docker build from Makefile

**Comment by calbrecht** [comment]
The `field-data` does look suspicious.

I might be tempted to give it a try fixing, and also did some research in the existing codebase and fixed issues.

Being completely new to vuejs (have a good understanding of react and js though) and this codebase, i could need some help getting around.

@rijkvanzanten, could you give an oversight or abstract explanation of what needs to be done here?

```
<div class="tree-view" 
  template="{{label}}" 
  autofocus="false" 
  loading="false" 
  width="full" 
  type="alias" 
  field-data="[object Object]" 
  data-v-1d7230b5="" 
  data-v-175a2f1a="">
    <ul class="drag-area root drag" data-v-0de81f94="" ...
```

**Comment by azrikahar** [comment]
@calbrecht thanks for following up. I looked into this thinking that the recent fix I did for repeaters relates to this, but it ended up being a different thing since they're not even sharing the same interface.

Just to give an explanation on what happened, anytime you save/configure the Display Template in treeview interface (as seen in your [2] screenshot), it's saving it as `template`. However when you view it (screenshot [1]), it's trying to display `displayTemplate` instead of `template`, so it ends up being empty. Since it's empty, plus sometimes a user can opt to not configure any display template, the current codebase make the template fall back to the primary key field, which is the ID:

https://github.com/directus/directus/blob/057af2313c90a112b7e850650f4f1a3335bbf765/app/src/interfaces/list-o2m-tree-view/list-o2m-tree-view.vue#L112-L114
